src/app/rundown-view/components/offsetable-piece/offsetable-piece.component.ts

```diff
@@ -48,6 +51,15 @@ export class OffsetablePieceComponent implements OnChanges, OnDestroy {
     }
   }
 
+  public updatePositionInVideo(tooltipMetadata: TooltipMetadata): void {
+    const videoLengthInPixels: number = this.widthInPixels
+    const positionInVideoInPercent: number = Math.round((tooltipMetadata.horizontalOffsetInPixels / videoLengthInPixels) * 100)
+
+    const videoDurationWithoutPlayedDurationInMs: number = Math.max(this.partDuration - this.playedDurationForPartInMs, 0)
+
+    this.positionInVideoInMs = Math.round(this.playedDurationForPartInMs + (videoDurationWithoutPlayedDurationInMs * positionInVideoInPercent) / 100)
```

Comment:
Video length here might be a bit confusing, since it describes the displayed width of the piece. I don't think it is necessary to assign `this.widthInPixels` to a constant. It can be used directly.

And the upscaling to 100 can also be removed here, resulting in:
```typescript
    const positionInVideoInPercent: number = tooltipMetadata.horizontalOffsetInPixels / this.widthInPixels
    const videoDurationWithoutPlayedDurationInMs: number = Math.max(this.partDuration - this.playedDurationForPartInMs, 0)
    this.positionInVideoInMs = Math.round(this.playedDurationForPartInMs + videoDurationWithoutPlayedDurationInMs * positionInVideoInPercent)
```

Reply:
Updated